src/hardware/memory.cpp

```diff
@@ -596,26 +605,21 @@ void MEM_PreparePCJRCartRom()
 	}
 }
 
-static int determine_num_megabytes(const int num_megabytes_pref)
+static void check_num_megabytes(const int num_megabytes)
 {
-	// max of 63 MB solves problems with certain xms handlers
-	constexpr auto min_megabytes  = 1;
-	constexpr auto safe_megabytes = SAFE_MEMORY - 1;
-	constexpr auto max_megabytes  = MAX_MEMORY - 1;
-
-	const auto num_megabytes = std::clamp(num_megabytes_pref, min_megabytes, max_megabytes);
+	assert(num_megabytes >= MinMegabytes);
+	assert(num_megabytes <= MaxMegabytes);
 
-	if (num_megabytes_pref > max_megabytes) {
-		LOG_WARNING("MEMORY: Memory size of %d MB is beyond the maximum of %d MB, limiting",
-		            num_megabytes_pref,
-		            max_megabytes);
-		assert(num_megabytes == max_megabytes);
+	if (num_megabytes > SafeMegabytesDos) {
+		LOG_WARNING("MEMORY: Memory sizes above %d MB aren't recommended for most DOS games",
+		            SafeMegabytesDos);
 	}
-
-	if (num_megabytes > safe_megabytes) {
-		LOG_WARNING("MEMORY: Memory sizes above %d MB aren't recommended", safe_megabytes);
+	if (num_megabytes > SafeMegabytesWin95) {
+		LOG_WARNING("MEMORY: Memory sizes above %d/%d MB aren't compatible with unpatched Windows 95/98, respectively",
+		            SafeMegabytesWin95,
+		            SafeMegabytesWin98);
+		// Limitation can be lifted with PATCHMEM by Rudolph R. Loew
```

Comment:
These logging improvements are excellent @FeralChild64; these showcase the type of user-friendly thinking we want to encourage in all code contributions ❤️